src-tauri/src/manager.rs

```diff
@@ -29,28 +38,66 @@ pub enum WatcherMessage {
 #[derive(Debug)]
 pub struct ManagerState {
     pub watchers_running: HashMap<String, bool>,
+    pub watchers_pid: HashMap<String, u32>,
 }
 
 impl ManagerState {
     fn new() -> ManagerState {
         ManagerState {
             watchers_running: HashMap::new(),
+            watchers_pid: HashMap::new(),
         }
     }
-    fn started_watcher(&mut self, name: &str) {
+    fn started_watcher(&mut self, name: &str, pid: u32) {
         println!("started {name}");
         self.watchers_running.insert(name.to_string(), true);
+        self.watchers_pid.insert(name.to_string(), pid);
         println!("{:?}", self.watchers_running);
     }
     fn stopped_watcher(&mut self, name: &str) {
         println!("stopped {name}");
         self.watchers_running.insert(name.to_string(), false);
+        self.watchers_pid.remove(name);
+    }
+    pub fn stop_watchers(&mut self) {
+        for (name, pid) in self.watchers_pid.iter() {
```

Comment:
We should make sure we aren't sending SIGTERM to any PIDs that belong to already-stopped watchers, as the PIDs might get reused after we've killed and `wait`ed them.

After looking at it again, this should be fine since we are removing the entry in the watchers_pid when watcher is stopped.